pkg/util/spanlogger/spanlogger.go

```diff
@@ -0,0 +1,39 @@
+package spanlogger
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/spanlogger"
+
+	util_log "github.com/cortexproject/cortex/pkg/util/log"
+
+	"github.com/grafana/loki/pkg/tenant"
+)
+
+const (
+	// TenantIDsTagName is the tenant IDs tag name.
+	TenantIDsTagName = spanlogger.TenantIDsTagName
+)
+
+// SpanLogger unifies tracing and logging, to reduce repetition.
+type SpanLogger = spanlogger.SpanLogger
+
+// New makes a new SpanLogger with a log.Logger to send logs to. The provided context will have the logger attached
+// to it and can be retrieved with FromContext.
+func New(ctx context.Context, method string, kvps ...interface{}) (*SpanLogger, context.Context) {
+	return spanlogger.New(ctx, util_log.Logger, method, tenant.DefaultResolver, kvps...)
```

Comment:
nit: in the `cortex` implementation of `spanklogger`, the `New` method actually uses `NewWithLogger`.

```suggestion
	return NewWithLogger(ctx, util_log.Logger, method, tenant.DefaultResolver, kvps...)
```

Reply:
@DylanGuedes I don't see they do? (https://github.com/grafana/mimir/blob/main/pkg/util/spanlogger/spanlogger.go). 
May be I didn't understand your suggestion?

Reply:
Ah. I understand what you mean. You mean from the `cortex` code. [here](https://github.com/cortexproject/cortex/blob/master/pkg/util/spanlogger/spanlogger.go#L40).

We actually don't need that, because, `dskit` doesn't have any such API now, 
https://github.com/grafana/dskit/blob/main/spanlogger/spanlogger.go#L41

Reply:
Dylan is saying you can use the function line 29 instead of the line 25.

Reply:
Ah. thanks @cyriltovena.

I didn't do that because, We don't need that in Loki now .`New()` already uses our `util_logger`(which is all our current loki code uses).

I added `NewWithLogger` if in future someone want's to pass logger explicitly.